lib/auth/init.go

```diff
@@ -724,7 +740,13 @@ func initializeAuthPreference(ctx context.Context, asrv *Server, newAuthPref typ
 
 		if !shouldReplace {
 			if os.Getenv(teleport.EnvVarAllowNoSecondFactor) != "true" {
-				return trace.Wrap(modules.ValidateResource(storedAuthPref))
+				err := modules.ValidateResource(storedAuthPref)
+				if errors.Is(err, modules.ErrCannotDisableSecondFactor) {
```

Comment:
This looks like a user-facing message, which is exactly what the "UserMessage" in a trace error is for.

Instead of updating the underlying error, would it make more sense to just augment it with a user message?

```
return trace.Wrap(modules.ValidateResource(storedAuthPref), userMessage) 
```